### PR TITLE
[Feature store] new features, enhancements, fixes, and tests

### PR DESCRIPTION
**Pull request by yaronha**
* extend classes to support Spark (and Pandas) dataframes
* add Spark, Mlrun Job, and Serving as ingestion options 
* enhance/fix merger and add remote job merger 
* add Parquet, HTTP, and Custom sources + allow setting key & time columns in source
* add Stream and Custom targets 
* implement IO using fsspec (works with all remote data stores like S3, v3io, http, azure blob)
* support inline code in feature sets & function reference (needed for UI)
* various bug fixes and new tests

### CODE REVIEW

File: mlrun/config.py [hunk 1]
```diff
@@ -146,6 +146,8 @@
             "nosql": "v3io:///projects/{project}/fs/{kind}",
         },
         "default_targets": "parquet,nosql",
+        "default_job_image": "mlrun/mlrun",
+        "default_spark_image": "",
```
Comment (Hedingber) [inline mlrun/config.py:150]: We already have `spark_app_image` and `spark_app_image_tag` do we really need this ?

File: mlrun/feature_store/retrieval/job.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    timestamp_column=None,
+    local=None,
+    watch=None,
+    function=None,
+    secrets=None,
+    auto_mount=None,
+):
+    name = vector.metadata.name
+    if not name:
+        raise ValueError("feature vector name must be specified")
```
Comment (Hedingber) [inline mlrun/feature_store/retrieval/job.py:22]: Use our errors, same for other places

File: tests/feature-store/test_feature_store.py [hunk 3]
```diff
@@ -155,36 +157,102 @@ def test_feature_set_db():
...
+
+    feature_set = fs.FeatureSet(
+        name, entities=[fs.Entity(entity)], timestamp_key=timestamp_key
+    )
+    feature_set.set_targets()
+    df = fs.ingest(feature_set, df_source, infer_options=fs.InferOptions.default())
+    return feature_set, df
+
+
+@pytest.mark.skipif(not has_db(), reason="no db access")
```
Comment (Hedingber) [inline tests/feature-store/test_feature_store.py:209]: Doesn't have to be part of this PR, but I've added an integration tests (SDK + API) infrastructure, currently all of these tests doesn't run automatically, you should move all of them there, it's under `tests/integration/sdk_api`

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: tests/feature-store/test_feature_store.py (hunk 3 above)
Comment (yaronha) [reply tests/feature-store/test_feature_store.py:209]: will do in the next PR

File: mlrun/feature_store/retrieval/job.py (hunk 2 above)
Comment (yaronha) [reply mlrun/feature_store/retrieval/job.py:22]: done

File: mlrun/config.py (hunk 1 above)
Comment (yaronha) [reply mlrun/config.py]: done, left the job one since it may need extra packages over mlrun/mlrun

**Review by Hedingber — APPROVED** [review]

**Pull request merged**